cmd/operator/main.go

```diff
@@ -282,20 +290,27 @@ func Main() int {
 		return 1
 	}
 
-	err = cc.CheckPrerequisites(ctx,
-		namespaces(cfg.Namespaces.AllowList).asSlice(),
-		verbs,
-		monitoringv1alpha1.SchemeGroupVersion.String(),
-		monitoringv1alpha1.PrometheusAgentName)
+	scrapeConfigSupported, err := checkPrerequisites(ctx, logger, cc, allowedNamespaces, verbs, monitoringv1alpha1.SchemeGroupVersion.String(), monitoringv1alpha1.ScrapeConfigName)
+	if err != nil {
+		cancel()
+		return 1
+	}
+
+	po, err := prometheuscontroller.New(ctx, cfg, log.With(logger, "component", "prometheusoperator"), r, scrapeConfigSupported)
+	if err != nil {
+		fmt.Fprintln(os.Stderr, "instantiating prometheus controller failed: ", err)
+		cancel()
+		return 1
+	}
 
-	switch {
-	case errors.Is(err, k8sutil.ErrPrerequiresitesFailed):
-		level.Warn(logger).Log("msg", "Prometheus agent controller disabled because prerequisites not met", "err", err)
-	case err != nil:
-		level.Error(logger).Log("msg", "failed to check prerequisites for prometheus-agent controller ", "err", err)
+	prometheusAgentSupported, err := checkPrerequisites(ctx, logger, cc, allowedNamespaces, verbs, monitoringv1alpha1.SchemeGroupVersion.String(), monitoringv1alpha1.PrometheusAgentName)
```

Comment:
(suggestion) a bit more readable and consistent with the other call
```suggestion
	prometheusAgentSupported, err := checkPrerequisites(
		ctx,
		logger,
		cc,
		allowedNamespaces,
		map[string][]string{
			monitoringv1alpha1.PrometheusAgentName:                           {"get", "list", "watch"},
			fmt.Sprintf("%s/status", monitoringv1alpha1.PrometheusAgentName): {"update"},
		},
		monitoringv1alpha1.SchemeGroupVersion.String(),
		monitoringv1alpha1.PrometheusAgentName,
	)
```